Upstream merge with PX4

kshitijgoel007 opened the pull request:
On the HITL firmware branch. Doing this for Notch filter and other latest features.

mcorah commented:
Is anything happening with this pull request? It might be a better idea to pull master into a branch here so that we have something static that we can control.

Also, for anyone else getting way too many notifications from this: you can unsubscribe from the pull request.

mrivi committed:
MPC_SPOOLUP_TIME -> COM_SPOOLUP_TIME

potaito committed:
New follow-me flight task

rename follow_me_status to follow_target_status

enable follow_target_estimator on skynode

implement the responsiveness parameter:
The responsiveness parameter should behave similarly to the previous
follow-me implementation in navigator. The difference here is that
there are now two separate gains for position and velocity fusion.
The previous implemenation in navigator had no velocity fusion.

Allow follow-me to be flown without RC

SITL tests for follow-me flight task

This includes:
- Testing the setting for the follow-me angle
- Testing that streaming position only or position
  and velocity measurements both work
- Testing that RC override works

Most of these tests are done with a simulated model
of a point object that moves on a straight line. So
nothing too spectacular. But it makes the test checks
much easier.

Since the estimator for the target actually checks new
measurements and compares them to old ones, I also added
random gausian noise to the measurements with a fixed seed
for deterministic randomness. So repeated runs produce
exactly the same results over and over.

Half of the angles are still missing in MAVSDK. Need to create
an upstream PR to add center left/right and rear left/right options.
These and the corresponding SITL tests need to be implemented
later.

sitl: Increase position tolerance during follow-me

Astro seems to barely exceed the current tolerance (4.3 !< 4.0)
causing CI to fail. The point of the CI test is not to check
the accuracy of the flight behaviour, but only the fact that the
drone is doing the expected thing. So the exact value of this
tolerance is not really important.

follow-me: gimbal control in follow-me

follow-me: create sub-routines in flight task class

follow-me: use ground-dist for emergency ascent

dist_bottom is only defined when a ground facing distance sensor exist.
It's therefore better to use dist_ground instead, which has the distance
to the home altitude if no distance sensor is available.

As a consequence it will only be possible to use follow-me in a valley
when the drone has a distance sensor.

follow-me: point gimbal to the ground in 2D mode

follow-me: another fuzzy msg handling for the estimator

follow-me: bugfix in acceleration saturation limit

follow-me: parameter for filter delay compensation

mantis: dont use flow for terrain estimation

follow-me: default responsiveness 0.5 -> 0.1

0.5 is way too jerky in real and simulated tests.

flight_task: clarify comments for bottom distance

follow-me: minor comment improvement

follow-me: [debug] log emergency_ascent

follow-me: [debug] log gimbal pitch

follow-me: [debug] status values for follow-me estimator

follow-me: setting for gimbal tracking mode

follow-me: release gimbal control at destruction

mavsdk: cosmetics :lipstick:

potaito committed:
failure_detector: add motor/ESC failure detection

potaito committed:
control_allocator: remove failed motor from effectiveness

- limit to handling only 1 motor failure
- Log which motor failures are handled
- Remove motor from effectiveness matrix without
  recomputing the scale / normalization

potaito committed:
simulator_mavlink: Add ESC telemetry

potaito committed:
Add SITL tests for control allocation

MaEtUgR committed:
battery: allow setting precise internal resistance with parameter

MaEtUgR committed:
battery: default to reasonable internal resistance

instead of disabling the load compensation using current.

MaEtUgR committed:
battery: only compensate using sane current

MaEtUgR committed:
battery: rather undercompensate for load

than over compensate which makes the estimate to high and breaks batteries
beause they get flown for too long.

potaito committed:
disable engine failure detector for SITL VTOL

Tried increasing ESC timeout for VTOL first

VTOL sitl tests are failing because the ESC telemetry seems to be coming
in at 0.3 Hz

potaito committed:
sitl: loosen some timeouts

The typhoon_h480 model would not always complete takeoff in 30 seconds
or finish the mission within 60 seconds.

sfuhrer committed:
VTOL: move to cpp params API

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
Commander: remove permanent stabilization option for fixed-wing flight

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
vtol_vehicle_status: replace several status bools by single vehicle_vtol_state

Replace vtol_in_rw_mode, vtol_in_trans_mode, in_transition_to_fw by vehicle_vtol_state.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

bkueng committed:
mavsdk_tests: ensure motor is stopped for motor failure test

bkueng committed:
esc_report: add actuator_function

bkueng committed:
failure_detector: add failure injector class for motor telemetry

This allows to test motor failures via 'failure motor off -i 1' on a real
system.

sfuhrer committed:
VTOL: remove some unsued variables

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

junwoo091400 committed:
FollowMe : Replace First order target position filter with Second order position and velocity filter

Follow me : tidied second order filter implementation

Added velocity filtered info to uORB follow target status message, and rebase to potaito/new-follow-me-task

FollowMe : Rebasing and missing definition fixes on target position second order filter

Follow Me : Remove Alpha filter delay compensation code, since second order filter is used for pose filtering now

Followme : Remove unused target pose estimation update function

Follow Target : Added Target orientation estimation logic

Follow Target : Replaced offset vector based setpoint to rate controlled orbital angle control

Follow Target : Bug fixes and first working version of rate based control logic, still buggy

Follow Target : Added target orientation, follow angle, orbit angle value into follow_target_status uORB message for debugging

Follow Target : Fix orbit angle step calculation typo bug

Follow Target : Few more fixes

Follow Target : Few fixes and follow angle value logging bug fix

Follow Target : Added lowpass alpha filter for yaw setpoint filtering

Follow Target : Remove unused filter delay compensation param

Follow Target : Add Yaw setpoint filter initialization logic and bufix for when unwrap had an input of NAN angle value

Follow Target : Add Yaw setpoint filtering enabler parameter

Follow Target : Change Target Velocity Deadzone to 1.0 m/s, to accomodate walking speed of 1.5 m/s

Follow Target : Add Orbit Tangential Velocity calculation for velocity setpoint and logging uORB topics

Follow target : Fix indentation in yaw setpoint filtering logic

Follow Target : Fix Follow Target Estimator timeout logic bug that was making the 2nd order pose filter reset to raw value every loop

Follow Target : Remove debug printf statement for target pose filter reset check

Follow Target : Add pose filter natural frequency parameter for filter testing

Follow Target : Make target following side param selectable and add target pose filter natural frequency param description

Follow Target : Add Terrain following altitude mode and make 2D altitude mode keep altitude relative to the home position, instead of raw local position origin

Follow Target : Log follow target estimator & status at full rate for filter characteristics test

Follow Target : Implementing RC control user input for Follow Target control

Follow Target : edit to conform to updated unwrap_pi function

Follow Target : Make follow angle, distance and height RC command input configurable

Follow Target : Make Follow Target not interruptable by moving joystick in Commander

Follow Target : reconfigure yaw, pitch and roll for better user experience in RC adjusting configurations, and add angular rate limit taking target distance into account

Follow Target : Change RC channels used for adjustments and re-order header file for clarity

Follow Target : Fix Parameters custom parent macro, since using DEFINE_PARAMETERS alone misses the parameter updates of the parent class, FlightTask

Follow Target : Fix Orbit Tangential speed actually being angular rate bug, which was causing a phenomenon of drnoe getting 'dragged' towards the target velocity direction

Follow Target : Final tidying and refactoring for master merge - step 1

Add more comments on header file

Follow Target : tidy, remove unnecessary debug uORB elements from follow_target_status message

Follow Target : Turn off Yaw filtering by default

Follow Target : Tidy maximum orbital velocity calculation

Follow Target : add yaw setpoint filter time constant parameter for testing and fix NAV_FT_HT title

Follow Target : Add RC adjustment window logic to prevent drone not catching up the change of follow target parameters

Follow Target : fixes

Follow Target: PR tidy few edits remove, and update comments

Follow Target : apply comments and reviews

Follow Target : Edit according to review comments part 2

Follow Target : Split RC adjustment code and other refactors

- Splitted the RC adjustment into follow angle, height and distance
- Added Parameter change detection to reset the follow properties
- Added comments and removed yaw setpoint filter enabler logic

Follow Target : Modify orbit angle error bufferzone bug that was causing excessive velocity setpoints when setpoint catched up with raw orbit setpoint

Follow Target : Remove buffer zone velocity ramp down logic and add acceleration and rate limited Trajectory generation library for orbit angle and velocity setpoint

Follow Target : Remove internally tracked data from local scope function's parameters to simplify code

Follow Target : Fix to track unwrapped orbit angle, with no wrapping

Follow Target : Apply user adjustment deadzone to reduce sensitivity

Follow Target : Apply suggestions from PR review round 2 by @potaito

Revert submodule update changes, fall back to potaito/new-followme-task

Follow Target : [Debug] Expose max vel and acceleration settings as parameters, instead of using Multicopter Position Controller
's settings

Follow Target : Use matrix::isEqualF() function to compare floats

Follow Target : Add Acceleration feedback enabler parameter and Velocity ramp in initial commit for overshoot phenomenon improvement

Follow Target : Implement Velocity feed forward limit and debug logging values

Follow Target : Apply Velocity Ramp-in for Acceleration as well & Apply it to total velocity setpoint and not just orbit tangential velocity component

Follow Target : Don't set Acceleration setpoint if not commanded

Follow Target : Use Jerk limited orbit angle control. Add orbit angle tracking related uORB values"

Follow Target : Add Orbit Angle Setpoint Rate Tracking filter, to take into consideration for calculating velocity setpoint for trajectory generator for orbit angle

Revert "Follow Target : Add Orbit Angle Setpoint Rate Tracking filter, to take into consideration for calculating velocity setpoint for trajectory generator for orbit angle"

This reverts commit a3f48ac7652adb70baf3a2fed3ea34d77cbd6a70.

Follow Target : Take Unfiltered target velocity into acount for target course calculation to fix overshoot orbit angle 180 deg flip problem

Follow Target : Remove Yaw Filter since it doesn't make a big difference in yaw jitterness

Follow Target : Remove velocity ramp in control & remove debug values from follow_target_status.msg

Follow Target : Tidy Follow Target Status message logging code

Follow Target : Remove jerk and acceleration settings from Follow Target orbit trajectory generation

Follow Target : Change PublicationMulti into Publication, since topics published are single instances

Follow Target : Edit comments to reflect changes in the final revision of Follow Target

Follow Target : Apply incorrectly merged conflicts during rebase & update Sticks function usage for getThrottled()

Follow Target : Apply final review comments before merge into Alessandro's PR

Apply further changes from the PR review, like units

Use RC Sticks' Expo() function for user adjustments to decrease sensitivity around the center (0 value)

Update Function styles to lowerCamelCase

And make functions const & return the params, rather than modifying them
internally via pointer / reference

Specify kFollowPerspective enum as uint8_t, so that it can't be set to negative value when converted from the parameter 'FLW_TGT_FP'

Fix bug in updateParams() to reset internally tracked params if they actually changed.

Remove unnecessary comments

Fix format of the Follow Target code

Fix Follow Perspective Param metadata

follow-me: use new trajectory_setpoint msg

Convert FollowPerspective enum into a Follow Angle float value

1. Increases flexibility in user's side, to set any arbitrary follow
angle [deg]
2. Removes the need to have a dedicated Enum, which can be a hassle to
make it match MAVSDK's side
3. A step in the direction of adding a proper Follow Mode (Perspective)
mode support, where we can support kite mode (drone behaves as if it is
hovering & getting dragged by the target with a leash) or a constant
orbit angle mode (Drone always on the East / North / etc. side, for
cinematic shots)

Continue fixing Follow Target MAVSDK code to match MAVSDK changes

- Support Follow Angle configuration instead of Follow Direction
- Change follow position tolerance logic to use the follow angle
*Still work in progress!

Update Follow Me MAVSDK Test Code to match MAVSDK v2 spec

- Add RC Adjustment Test case
- Change follow direction logic to follow angle based logic completely
- Cleanup on variable names and comment on code

follow-me: disable SITL test

Need to update MAVSDK with the following PR:
https://github.com/mavlink/MAVSDK/pull/1770

SITL is failing now because the follow-me
perspectives are no longer defined the
same way in MAVSDK and in the flight task.

update copyright year

follow-me: mark uORB topics optional

Apply review comments

more copyright years

follow-me sitl test: simpler "state machine"

flight_mode_manager: exclude AutoFollowTarget and Orbit on flash contrained boards

Remove unnecessary follow_target_status message properties

- As it eats up FLASH and consumes uLog bandwidth

dagar committed:
ekf2: request mag yaw reset after calibration or sensor change

MaEtUgR committed:
Use signNoZero() where possible

dagar committed:
ekf2: estimator aid source status (starting with fake position)

dagar committed:
ekf2: estimator aid source status (GPS pos, GPS vel)

dagar committed:
ekf2: yaw estimator add yaw_composite_valid boolean

dagar committed:
ekf2: EKFGSF_yaw delete unnecessary internal state

sfuhrer committed:
Set tailsitter flag via vehicle status

Removes the necessity of including vtol_type.h in other modules.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

mcsauder committed:
Replace EKF/common.h #defines with enums.

sfuhrer committed:
VTOL: rename params _PTCH to _PITCH

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

mcsauder committed:
Add gyro clipping to mirror accel clipping monitoring instances.

davids5 committed:
px4_fmu-v6x:Add Rev 4 Sensors

davids5 committed:
px4_fmu-v5X:Added Holybro mini base board

bkueng committed:
ActuatorEffectivenessRotors: fix motor count check

Check during init to avoid out-of-bound access.

bkueng committed:
control_allocator: increase max num motors to 12

bkueng committed:
control_allocator: add helicopter mixer

Same logic as the existing mixer.
Untested.

Jaeyoung-Lim committed:
Add omnicopter SITL model

F

MaEtUgR committed:
rc_update: add unit tests for mode slot

To verify RC mode switch and mode button functionality works as expected.

bkueng committed:
failure_detector: allow disabling attitude failure (as already documented)

bkueng committed:
mc_rate_control: pass through 3D thrust

bkueng committed:
control_allocator: show motor axis for MC (as advanced)

dagar committed:
boards: px4_fmu-v5/v5x init all I2C pins

bkueng committed:
cuav/x7pro/test.px4board: disable module to reduce flash

bkueng committed:
cubepilot/cubeorange/test.px4board: disable module to reduce flash

bkueng committed:
holybro/kakutef7: disable module to reduce flash

bkueng committed:
px4/fmu-v2/multicopter.px4board: disable module to reduce flash

bkueng committed:
px4/fmu-v5/protected.px4board: disable module to reduce flash

junwoo091400 committed:
Add get*Expo() functions to the Sticks library

To return Exponential Values, which is helpful for reducing the
sensitivity of the stick around the centered value (0), since it's
exponential curve.

Useful for user adjustment implementations, where accidentally touching the stick
wouldn't have so much effect when using the Exponential value, compared
to using the raw position value.

sfuhrer committed:
Control Allocation: Tiltrotor: pass only thrust magnitude instead of 3D thrust to allocator

Special case tiltrotor: instead of passing a 3D thrust vector (that would mostly have a x-component in FW, and z in
MC), pass the vector magnitude as z-component, plus the collective tilt. Passing 3D thrust plus tilt is not feasible as they
can't be allocated independently, and with the current controller it's not possible to have collective tilt calculated
by the allocator directly.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

Commit:
fw pos ctrl: calculate control interval once

- Use the same time interval for all position control logic (including TECS)
- Sync naming in control methods
- Remove some unused input arguments

Commit:
fw pos ctrl: only manipulate heading hold yaw in manual position control mode

Commit:
fw pos ctrl: centralize parameter and state resets

Commit:
fw pos ctrl: some incremental clean up of the class
- documentation of units, params, returns, and descriptions for variables and methods
- rename ambiguous or erroneous state variables
- remove unused or unecessary input arguments to functions
- remove ugly header white space

taileron committed:
boards: matek_h743-mini sync all Matek H743 boards

taileron committed:
boards: matek h743 update unified target

Igor-Misic committed:
uavcan: use timer 6 by default on stm32f7

KonradRudin committed:
Remove contradicting geofence parameter description

The flight termination action on geofence violation is described as only trigger, when the corresponding circuit breaker is not disabled. However, the description of the circuit breaker states, that the geofence action is not depedning on this circuit breaker. The implementation follows the description of the circuit breaker. Hence the GF_ACTION description is adapted.

sfuhrer committed:
Control Allocation: add option to disable yaw by differential thrust

This replaces the propeller_torque_disabled flag to disable yaw by differential thrust
for tiltrotor and tailsitter VTOLs, as propeller_torque_disabled is not enough to set
effectiveness of an acutator in the yaw axis to 0 in case it's tilted.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
ROMFS: SITL tiltrotor: enable control allocaton and only tilt front motors

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

dagar committed:
boards: modalai_fc-v1 enable dshot on all outputs

bkueng committed:
log_writer_file: protect access to _should_run, use px4::atomicbool for _exit_thread

bkueng committed:
log_writer_file: fix corner case when mission log is enabled

Normally _should_run for mission is only ever true if _should_run for the
normal log is. There are exceptions though:
- the log buffer fails to allocate
- there was a write failure (e.g. due to SD card removal)
In that situation, the writer would not wait anymore but busy-loop.

MaEtUgR committed:
Separate message for trajectory setpoint

sfuhrer committed:
Commander: add max flight time RTL

Adds COM_FLT_TIME_MAX param and logic in Commander to enforce RTL when
flight time is above this value. User can only override to LAND mode,
but not proceed flight beyond that.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
Commander: high wind speed handling updates

- add logic for detecting high wind speed in Commander,
and handle it toghether with wind speed warning
- trigger and enforce RTL if COM_WIND_MAX is breached

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

MaEtUgR committed:
mavlink_receiver: check entire Vectors for NAN

Note: Behavioral change Inf also results in the flag being true.

dagar committed:
cmake: NuttX check that CONFIG_ARCH_BOARD_CUSTOM_DIR is in PX4_BOARD_DIR

MaEtUgR committed:
uuv_pos_control: siplify passing on trajectory position setpoint

MaEtUgR committed:
vehicle_local_position_setpoint: reorder fields for clarity

PetervdPerk-NXP committed:
Upgrade libcanard to 3.0 and rename uavcan_v1 to cyphal
Did some prep work for redundant interfaces by introducing CanardHandle class to decouple physical interfaces from cyphal.cpp

bkueng committed:
omnibus/f4sd/ekf2: disable module to reduce flash

bkueng committed:
px4/fmu-v2/fixedwing: disable module to reduce flash

bkueng committed:
px4/fmu-v5/stackcheck: disable module to reduce flash

PX4BuildBot committed:
Update submodule mavlink to latest Tue May 10 12:39:00 UTC 2022

    - mavlink in PX4/Firmware (d6dff7daa984514185a8f31f6e653a69a278ded8): https://github.com/mavlink/mavlink/commit/3b52eac09c2e37325e4bc49cd2667ea37bf1d7d2
    - mavlink current upstream: https://github.com/mavlink/mavlink/commit/a1cb2c0e71f191f04da3d92d92db8702a7e91ff6
    - Changes: https://github.com/mavlink/mavlink/compare/3b52eac09c2e37325e4bc49cd2667ea37bf1d7d2...a1cb2c0e71f191f04da3d92d92db8702a7e91ff6

    a1cb2c0e 2022-05-05 KonradRudin - Figure eight MAV_CMD (#1831)
0ebdf846 2022-04-28 Hamish Willee - common.xml: Deprecate MAV_CMD_GET_MESSAGE_INTERVAL for MAV_CMD_REQUEST_MESSAGE. (#1835)
f357fc78 2022-04-27 Hamish Willee - Deprecate MAV_CMD_GET_HOME_POSITION for MAV_CMD_REQUEST_MESSAGE (#1834)

dagar committed:
boards: px4_fmu-v5_uavcanv0periph disable common differential pressure sensors to save flash

dagar committed:
Jenkins: compile update uavcanv1 -> cyphal

dagar committed:
boards: matek_h743_bootloader fix NuttX CONFIG_ARCH_BOARD_CUSTOM_DIR

dagar committed:
boards: matek_h743-mini_bootloader fix NuttX CONFIG_ARCH_BOARD_CUSTOM_DIR

dagar committed:
Jenkins: compile add all matek targets and sort

dagar committed:
github actions: add all matek targets and sort

dagar committed:
Update submodule GPSDrivers to latest Tue May 10 12:38:49 UTC 2022

    - GPSDrivers in PX4/Firmware (3ac8fdbe2901dcd812c8e1944f72765ddff4c022): https://github.com/PX4/PX4-GPSDrivers/commit/6534b050ee1a48af7932c46a9a87277eed1cc997
    - GPSDrivers current upstream: https://github.com/PX4/PX4-GPSDrivers/commit/58968922b718176be8756f11113d16b2cfbc4022
    - Changes: https://github.com/PX4/PX4-GPSDrivers/compare/6534b050ee1a48af7932c46a9a87277eed1cc997...58968922b718176be8756f11113d16b2cfbc4022

    5896892 2022-04-26 Seppe Geuens - sbf: add heading support (#100)

Co-authored-by: PX4 BuildBot <bot@px4.io>

dagar committed:
Makefile: bootloaders_update add all Matek targets

dagar committed:
.vscode/.gitignore ignore all log files

dagar committed:
ROMFS: delete steadidrone airframes

dagar committed:
ROMFS: delete minimal DJI airframes

dagar committed:
ROMFS: delete 13010_claire airframe

dagar committed:
ROMFS: delete 3030_io_camflyer airframe

dagar committed:
ROMFS: delete 3034_fx79 airframe

dagar committed:
ROMFS: delete 3035_viper airframe and mixer

dagar committed:
ROMFS: delete 3037_parrot_disco_mod airframe

PetervdPerk-NXP committed:
Cyphal restore O1Heap statistics

PetervdPerk-NXP committed:
Update UCANS32K146 clocking to use XTAL and support higher pheriphal freq

dagar committed:
commander: remove compile time dependencies on non-commander parameters

 - this allows builds with mavlink fully disabled
 - move commander MAN_ARM_GESTURE, RC_MAP_ARM_SW, MC_AIRMODE checks to manual_control

dagar committed:
drivers/rc_input: ensure RC inversion is disabled initially and latch RC_INPUT_PROTO conservatively

 - this allows jumping straight to a non-SBUS RC protocol
 - increased the scan time per protocol 300->500 ms, which the newer DSM parser seems to need in some cases.
 - only set RC_INPUT_PROTO if we've had a successful RC lock for > 3 seconds

AlexKlimaj committed:
Fix uavcan battery causing immediate RTL time remaining low

hamishwillee committed:
pwm.cpp: remove the test example (#19587)

bkueng committed:
output drivers: init SmartLock after exit_and_cleanup

This fixes an invalid memory access when exiting the module:
exit_and_cleanup destroys the object, but lock_guard is destructed after
and accesses the lock.

bkueng committed:
dshot: handle DSHOT_MIN for reversible outputs

Also ensures there's no deadband if dead_l == dead_h (the default).

MaEtUgR committed:
trajectory_setpoint: correct comment typo "kinematically"

Co-authored-by: Mathieu Bresciani <brescianimathieu@gmail.com>

MaEtUgR committed:
mavlink_mission: add more specific information to the error message

MaEtUgR committed:
Commander: don't disarm on landing amid a mission

taileron committed:
boards: matek h743-mini specify drivers to still fit to flash

 - the code has become a bit bigger again that now the drivers have to be specified a bit more precisely to still fit into the flash.

MaEtUgR committed:
mission_block: make "Delay until" item work while landed

To enable landing, waiting and taking off again e.g. for delivery.

bresch committed:
baro bias: improve innovation sequence monitoring

The baro observation noise parameter is often over-estimated in order as
a measure to mitigate temporary offsets in the readings due to wind
gusts or poor pressure compensation tuning. The side effect is that the
innovation sequence monitoring based on normalized innovation struggles
to detect an offset in the state because the innovation isn't
statistically significant.
To counter this issue, a simpler check is added to trigger the process
noise boost when the innovation has the same sign for a long period of
time.

MaEtUgR committed:
ArmStateMachine: replace state name array with method

dagar committed:
drivers/imu/invensense/icm42670p: cleanup and small fixes

PetervdPerk-NXP committed:
UCANS32K146 Fix CAN1 STB pin

MaEtUgR committed:
ArmStateMachine: move arm state into the class

MaEtUgR committed:
ArmStateMachine: remove dependency on armed.armed

To have the internal state as single source of truth
for the arming state within Commander.

dagar committed:
icm42670p run at full speed

sfuhrer committed:
tailsitter: guard against division by 0

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
VTOL paras: add min of 0.1 to transition times

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

sfuhrer committed:
simulator_mavlink: fix MAV_TYPE_VTOL_TAILSITTER case

This type (23) doesn't specify a motor number, so it can't be properly handled.
There are duo (19) and quad (20) tailsitter types that still work in simulation.

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

PX4BuildBot committed:
Update submodule mavlink to latest Thu May 19 00:38:59 UTC 2022

    - mavlink in PX4/Firmware (87c73145b36a835b1635de0498a5613a7af5cafc): https://github.com/mavlink/mavlink/commit/a1cb2c0e71f191f04da3d92d92db8702a7e91ff6
    - mavlink current upstream: https://github.com/mavlink/mavlink/commit/99e82cad70494903a23a67de08ff9cbb5918d8f3
    - Changes: https://github.com/mavlink/mavlink/compare/a1cb2c0e71f191f04da3d92d92db8702a7e91ff6...99e82cad70494903a23a67de08ff9cbb5918d8f3

    99e82cad 2022-05-19 Hamish Willee - Deprecated GPS_INJECT_DATA (#1842)
bf3df03d 2022-05-16 Hamish Willee - WIND_COV - accuracy units are m/s (#1844)
a73d4864 2022-05-11 Hamish Willee - development.xml FIGURE_EIGHT_EXECUTION_STATUS - add a note about it being set at lower rate (#1841)

mcsauder committed:
Standardize variable naming and formatting across vehicle attitude controller files.

junwoo091400 committed:
Improve uLog message struct definitions

1. Rename *_header_s structs to *_s, since the _header postfix is not
helpful
2. Rename the "key" string variables in the message structs to
"key_value_str" as the string actually contains not just the key but the
key and value pair information
3. Add comments on other uLog messages to clarify use (need more
improvement / can be even more simplified)

junwoo091400 committed:
Update Issue and Pull Request Templates

1. Change the paragraph headings to proper Markdown headings (easier to
link / structure the Markdown)
2. Move PULL_REQUEST_TEMPLATE into .github folder
3. Change description in Issue template and remove outdated DevGuide
Repository information
4. Add a bug emoji to bug report isue template
5. Support automatically adding labels 'bug-report' and
'feature-request' to easily sort / filter appropriate issues in Github

potaito committed:
ubuntu.sh: fix gazebo and openjdk for 22.04

MaEtUgR committed:
commander_params: shorten low battery action delay

I got multiple times the feedback now that a consistent delay
is helpful and makes sense but the default delay
is too long
for low battery action where you're trying to come back in time
and possibly the emergency reaction kicks in while the critical action
is executing which leads to a longer accumulated delay.

Commit:
Commander: ignore MAV_CMD_REQUEST_MSG

This commit adds the MAV_CMD_REQUEST_MESSAGE to the list of vehicle
commands which are ignored without generating a warning sound.

Igor-Misic committed:
px4io: replace safety_off state with safety button event (#19558)

internal PX4IO safety_off state is removed and replaced with a normal safety button event. From this 'commit' commander is taking care of the PX4IO safety.

Igor-Misic committed:
Revert "px4io: replace safety_off state with safety button event (#19558)"

This reverts commit 12a81979a8e53b482f45b888faa1c8ed79d23ada.

dagar committed:
commander: publish actuator_armed first on any change

julianoes committed:
commander: lockdown is not termination

We use lockdown to prevent outputs like motors and servos from being
active in HITL simulation. This means that we can't treat the lockdown
flag as a flight_terminated, otherwise we can't arm in HITL at all.

NicolasM0 committed:
HITL: undefined time_remaining_s should be NAN

Igor-Misic committed:
commander/safety: replace safety.msg with Safety class (#19558)

julianoes committed:
sitl_gazebo: update submodule

This fixes the issue where HITL doesn't connect over USB.

davids5 committed:
boards: new px4_fmu-v6c board support (#19544)

dagar committed:
commander: handle RC loss timeout if manual_control_setpoint stops publishing (#19680)

Co-authored-by: Matthias Grob <maetugr@gmail.com>

PX4BuildBot committed:
Update submodule sitl_gazebo to latest Sat May 21 12:38:49 UTC 2022

    - sitl_gazebo in PX4/Firmware (422be90140c9dee6f0e59ee4d181301b13fb11de): https://github.com/PX4/PX4-SITL_gazebo/commit/48440d7b5c78a21182415266334981f1163f4b2c
    - sitl_gazebo current upstream: https://github.com/PX4/PX4-SITL_gazebo/commit/5eb5df80450a412076bfc24e7dd343839020f056
    - Changes: https://github.com/PX4/PX4-SITL_gazebo/compare/48440d7b5c78a21182415266334981f1163f4b2c...5eb5df80450a412076bfc24e7dd343839020f056

    5eb5df8 2022-04-28 stmoon - ROS2_PLUGIN set OFF by default
c8f4d5b 2022-04-28 stmoon - update the motor failure plugin for ros2

dagar committed:
Update world_magnetic_model to latest Sat May 21 11:14:10 UTC 2022

julianoes committed:
ROMFS: disable UAVCAN in HITL

Without this, uavcan creates MixingOutput classes which then create
empty actuator_outputs publications. This then prevents the motor
output in HITL to be forwarded to the simulator via mavlink.

dagar committed:
Jenkinsfile: hardware test rack skip runing HITL airframes

dagar committed:
boards: px4_fmu-v2_fixeding disable fw_autotune_attitude_control to save flash

dagar committed:
ekf2: estimator aid source status (baro height)

dagar committed:
ekf2: estimator aid source status (range height)

aksuns96 committed:
sensors/vehicle_magnetometer: fix multi_mode check

Signed-off-by: Serhat Aksun <serhat.aksun@maxwell-innovations.com>

dagar committed:
icm42688p: only check configured registers periodically (as intended)

MaEtUgR committed:
battery_params: increase default empty cellvoltage to 3.6v

Based on feedback that very often the battery is used down too low.
I observed this happens consistently when the cell voltage is properly
load compensated. The default load compensation before #19429 was very
inaccurate and resulted in unpredictable estimate.
After that if there is a usable current measurement and the battery is
within expected tolerances of the default internal resistance the
compensation is pretty good and 3.5V is too low for an empty compensated
cell voltage. That was seen in various logs where the compensated
cell voltage was already dropping fast after 3.6V.

In case the voltage is not load compensated the vehicle estimates the
state of charge a bit too low which is safer than to high
especially for a default configuration.

Igor-Misic committed:
Revert "px4iofirmware: fix for bug introduced with #19558"

This reverts commit 79a34b5aed8ee9eb10afda3ae83f1be02ad897a8.

Igor-Misic committed:
px4iofirmware: fix for bug introduced with #19558

bresch committed:
ekf2: use explicit flags instead of bitmask position

This prevents bitmask mismatch when a new flag is inserted

mcsauder committed:
rcS: Keep Mag Cal with AUTOCONFIG param reset in rcS, deprecate rcS AUTOCNF param (#19693)

* Deprecate the rcS AUTOCNF script param and associated logic.

* Update posix rcS to match previous commit.

bkueng committed:
MulticopterRateControl: fix thrust sign for acro

Regression from fbc109436f00a815eb6130c6d177086e6b1ae80a

bresch committed:
ekf2_post-processing: use estimator_status_flags instead of bitmasks

davids5 committed:
px4_fmu-v6c:Fix mag rotation

davids5 committed:
px4_fmu-v6X:Added Holybro mini base board

junwoo091400 committed:
uLog message definition comment improvements

- Added more comments
- Converted to DOxygen comment format for the comments on struct members

dagar committed:
Update submodule GPSDrivers to latest Wed May 25 12:38:58 UTC 2022

- devices in PX4/Firmware (7cc84e0f9e44697475b0f355df9a29de6ea6b9ff): https://github.com/PX4/PX4-GPSDrivers/commit/58968922b718176be8756f11113d16b2cfbc4022
    - devices current upstream: https://github.com/PX4/PX4-GPSDrivers/commit/181fae1a4b5e33576d786755782adb2f195ecc48
    - Changes: https://github.com/PX4/PX4-GPSDrivers/compare/58968922b718176be8756f11113d16b2cfbc4022...181fae1a4b5e33576d786755782adb2f195ecc48

    181fae1 2022-05-23 numan - README: add information for definition.h

MaEtUgR committed:
Commander: only initialize to tablet mode and don't switch back on RC loss

MaEtUgR committed:
Commander: differentiate clearly between to arm and to disarm transition

MaEtUgR committed:
Commander: switch back to Hold mode after autonomous landing

MaEtUgR committed:
PreFlightCheck: disallow arming in land mode

after the automatic switch to Hold mode was introduced.

davids5 committed:
uc_stm32h7_can:Correct initalization of the mumber of interfaces

   H7 Only supports 2 not 3 CAN interfaces.

   CanInitHelper passes in in the run time configuration of
   the number of interfaces. The code was ignoring these!

potaito committed:
Match commander uORB var names to message names (#19707)

* match vehicle_status

* match home_position

* match vehicle_command_ack

* match actuator_armed

* match vehicle_control_mode

* match commander_state

MaEtUgR committed:
Commander: fix overlooked rebase naming error

dagar committed:
boards: delete omnibus_f4sd_ekf2 due to flash overflow

davids5 committed:
flash_cache:Flush complete cache line

davids5 committed:
Update all H7 Bootloders

davids5 committed:
boards: STM32H7 pad to 256 bit - 32 bytes (#19724)

dagar committed:
boards: pixhawk 2 cube skip starting low quality l3gd20 gyro to save memory and cpu

 - free memory is getting tight on these older boards (depending on
configuratoin) and the pixhawk 2 cube still has 2 other superior IMUs, so this is just
dropping dead weight

dagar committed:
boards: reduce SPI DMA buffers on older STM32F4 boards

 - on common IMUs like the mpu6000, mpu9250, icm20602, etc each FIFO
sample is only 12 bytes so this is still more than large enough for the
worst case transfer

dlech committed:
vscode: remove terminal settings

These terminal settings are a matter of personal preference and should
not be part of the project configuration.

PX4BuildBot committed:
Update submodule mavlink to latest Sat May 28 00:38:55 UTC 2022

    - mavlink in PX4/Firmware (56106be4805cb2f70d22ad31b87fbd9774f0d0f3): https://github.com/mavlink/mavlink/commit/99e82cad70494903a23a67de08ff9cbb5918d8f3
    - mavlink current upstream: https://github.com/mavlink/mavlink/commit/0909631552187b998dd6359f998ee78ee8765728
    - Changes: https://github.com/mavlink/mavlink/compare/99e82cad70494903a23a67de08ff9cbb5918d8f3...0909631552187b998dd6359f998ee78ee8765728

    09096315 2022-05-26 Hamish Willee - Clarify and cross reference home position everywhere (#1836)
38c1a99f 2022-05-25 Julian Oes - WIND_COV: clarify description and unknown value (#1845)

dagar committed:
ekf2: sensor_simulator fix enum shadow

dagar committed:
lib/motion_planning: VelocitySmoothingTest delete set but not used variable

dagar committed:
mavlink: MavlinkStatustextHandlerTest ignore Wcast-align

dagar committed:
update NuttX and apps to latest with sem holder fixes and updated ostest

dagar committed:
boards: test boards add NuttX ostest and run on Jenkins test rack

dagar committed:
boards: NuttX update all boards to preallocated sem holder list

 - CONFIG_SEM_PREALLOCHOLDERS=32
 - CONFIG_SEM_NNESTPRIO=16 (default)

dagar committed:
Jenkins: reboot after ostest completes

 - temporary workaround to ostest apparently leaving the system in a bad state

bkueng committed:
control_allocator: disable backup schedule with lockstep

The backup schedule triggers updates too early and leads to simulator errors:
ERROR [simulator] poll timeout 0, 22

MaEtUgR committed:
FlightTask: Weather vane cleanup

Remove the entire external yaw handler, dynamic memory allocation,
pointer passing logic. Directly instanciate the weather vane instance
in the flight tasks that support it.

Igor-Misic committed:
px4io: replace safety_off state with safety button event (#19558)

internal PX4IO safety_off state is removed and replaced with a normal safety button event. From this 'commit' commander is taking care of the PX4IO safety.

MaEtUgR committed:
FlightModeManager: only execute flight task commands when multicopter

Igor-Misic committed:
drivers/px4io:  support legacy px4io firmware to enter BL

mcsauder committed:
attitude_estimator_q: Run() method refactoring (#19526)

 * Refactor attitude_estimator_q_main.cpp Run() method by breaking apart into separate method calls:
    * update_vehicle_local_position()
    * update_motion_capture_odometry()
    * update_visual_odometry()
    * update_magnetometer()
    * update_vehicle_attitude()
    * update_sensors()
 * Rename init_attitude_q()
 * Standardize whitespace formatting
 * Add remaining c++ style initializers.

tpwrules committed:
boards/mro/pixracerpro: fix voltage/current monitoring

This corrects the board definition to use the proper polarity
for the brick power valid signal, thus allowing the board to
detect the battery and monitor it properly.

BA-ED committed:
dshot: corrected DShot motor spin direction command

Some ESCs don't store DShot_cmd_spin_direction_reversed persistently

abarcis committed:
vehicle_command.msg: unneeded comment closings removed (#19759)

They cause problems with building px4_msgs in ROS2 Humble Hawksbill and removing them fixes the issue.

Co-authored-by: Agata Barcis <agata.barcis@tii.ae>

bkueng committed:
FailureDetector: check if ESCs have current

And increase the timeout to 300ms, as some ESCs only update with 10Hz.

bresch committed:
terrain est: clear innovation/var/test ration when aiding stops

bkueng committed:
failure: fix invalid memory access

failure_units[unit].key was incorrect

bkueng committed:
SITL: enable failure command (SYS_FAILURE_EN=1)

MaEtUgR committed:
Commander: remove system_sensors_initialized

and system_hotplug_timeout. They're not in use for over 2 years.
Instead control LED with preflight checks.

bkueng committed:
test_multicopter_failure_injection: remove 'Reject before arming' test

Not sure why injection should fail before arming.

PX4BuildBot committed:
Update world_magnetic_model to latest Sat Jun  4 11:14:10 UTC 2022

PX4BuildBot committed:
[AUTO COMMIT] update change indication

dagar committed:
sensors/vehicle_imu: don't bother checking IMU_GYRO_RATEMAX

dagar committed:
sensors/vehicle_angular_velocity: add IMU_GYRO_RATEMAX constraints

dagar committed:
Update submodule GPSDrivers to latest Sat Jun  4 12:38:49 UTC 2022 (#19766)

    - GPSDrivers in PX4/Firmware (6e77a084cd43830c8b13018b7fd5470da9bc4ff5): https://github.com/PX4/PX4-GPSDrivers/commit/181fae1a4b5e33576d786755782adb2f195ecc48
    - GPSDrivers current upstream: https://github.com/PX4/PX4-GPSDrivers/commit/016c37cd1f18c716427e2465d8daa6aa1054b0f1
    - Changes: https://github.com/PX4/PX4-GPSDrivers/compare/181fae1a4b5e33576d786755782adb2f195ecc48...016c37cd1f18c716427e2465d8daa6aa1054b0f1

    016c37c 2022-06-01 Julian Oes - sbf: fix overrun on invalid length

MaEtUgR committed:
MulticopterRateControl: use constructor to copy thrust setpoint array

MaEtUgR committed:
Commander: switch battery bitflied to dedicated datatype

bresch committed:
ekf2 preflight: only check innovation of active height sources

sfuhrer committed:
navigator_main: DO_REPOSITION: only trigger reposition setpoint update if vehicle is armed

Signed-off-by: Silvan Fuhrer <silvan@auterion.com>

MaEtUgR committed:
Style refactoring related to safety button

MaEtUgR committed:
Safety: keep initialized constant flags when safety disabled

MaEtUgR committed:
Commander: fix startup sound interrupted by safety button initialization

The safetyButtonHandler() reports that the safety statatus
changed on the first loop iteration when safety is disabled which makes
sense to inform the system that safety is off but the tune for the user
should not be played because it interrupts the startup tune.

MaEtUgR committed:
manual_control_params: configure arm gesture doesn't require reboot

I added wrong documentation here.

bresch committed:
wind_estimator: use noise spectral density for process noise tuning

The noise spectral density, NSD, (square root of power spectral density) is a
continuous-time parameter that makes the tuning independent from the EKF
prediction rate.
NSD corresponds to the rate at which the state uncertainty increases
when no measurements are fused into the filter.
Given that the current prediction rate of the wind estimator is 1Hz, the
same tuning is obtained with the same values as before.

bresch committed:
ekf2 wind: use noise spectral density for process noise tuning

The noise spectral density, NSD, (square root of power spectral density) is a
continuous-time parameter that makes the tuning independent from the EKF
prediction rate.
NSD corresponds to the rate at which the state uncertainty increases
when no measurements are fused into the filter.
Given that the current prediction rate of EKF2 is 100Hz, the
same tuning is obtained by dividing the std_dev legacy parameter by 10:
nsd = sqrt(std_dev^2 / 100Hz)

bresch committed:
WindEstimator: make wind process noise tuning same as EKF2

Commit:
fw pos ctrl: update method documentation

ThomasDebrunner committed:
SIH in SITL with lockstep (#19028)

* sih: Move sih out of work queue
This reverts commit bb7dd0cf0016ac1998a79814ec52b15079a74c1a.

* sih-sim: Enable sih in sitl, together with lockstep

* sih-sim: new files for sih: quadx and airplane

* sih: Added tailsitter for sih-sitl simulation

* sitl_target: Added seperate target loop for sih

* jmavsim_run: Allow jmavsim to run in UDP mode

* lockstep: Post semaphore on last lockstep component removed

* sih-sim: Added display for effectively achieved speed

* sih: increase stack size

* sih-sim: Improved sleep time computation, fixes bug of running too fast

* sitl_target: place omnicopter in alphabethic order

Co-authored-by: romain-chiap <romain.chiap@gmail.com>
Co-authored-by: Matthias Grob <maetugr@gmail.com>

MaEtUgR committed:
ArmStateMachine: simplify how preflight checks are called

bresch committed:
migrate wind process noise parameters

fury1895 committed:
Airspeed params: change default for ASPD_SCALE_APPLY to 2

PX4BuildBot committed:
Update submodule mavlink to latest Fri Jun 10 00:38:25 UTC 2022

    - mavlink in PX4/Firmware (59630bcc7e7e983db2d9138c8254144c77855adc): https://github.com/mavlink/mavlink/commit/0909631552187b998dd6359f998ee78ee8765728
    - mavlink current upstream: https://github.com/mavlink/mavlink/commit/05864e218e204f1ebeee5555988150fcddbd873e
    - Changes: https://github.com/mavlink/mavlink/compare/0909631552187b998dd6359f998ee78ee8765728...05864e218e204f1ebeee5555988150fcddbd873e

    05864e21 2022-06-09 Peter Barker - common.xml: add ignition_voltage to EFI_STATUS (#1854)
adc9c3f5 2022-06-01 olliw42 - update storm32.xml (nfc) (#1851)
49dbdb66 2022-06-01 Randy Mackay - ardupilotmega: add nav_attitude_time command (#1852)

davids5 committed:
px4_fmu-v6x:HB Mini add Ver 3, Ver 4 init

taileron committed:
disable uart´s dma

Still no way to get GPS and auto flash of the IO without disabling their uart´s dma

MaEtUgR committed:
Commander: move first preflight check run to constructor

Igor-Misic committed:
drivers/pwm_out: set default failsafe values

MaEtUgR committed:
Commander: execute pre arm check with preflight checks

Igor-Misic committed:
driver/px4io: set default failsafe values

Igor-Misic committed:
uavcan: don't print an error if there is no UAVCAN device on the CAN bus

MaEtUgR committed:
Commander: replace arm requirements

MaEtUgR committed:
WeatherVane: Allow weathervane on multirotors not just VTOLs

Igor-Misic committed:
boards/bitcraze: add PWM_SERVO_STOP define

taileron committed:
boards/diatone/mamba-f405-mk2: symmetric buffers for wifi telemetry (#19808)

Symmetric buffers allow a much more reliable QGC Wifi telemetry connection especially when (virtual) joysticks are used.  (this board does not provide RX DMA on UART 4 as its timer does DSHOT).

dagar committed:
ekf2: setEkfGlobalOrigin() reset baro and hgt sensor offsets if necessary

 - handle uninitalized _gps_alt_ref
 - add basic lat/lon/alt sanity checks

dagar committed:
boards: disable modules to save flash

 - px4_fmu-v5x_rtps disable common barometers to save flash
 - px4_fmu-v6x_default disable telemetry drivers to save flash

chris1seto committed:
Re-set param limits for fw tuning values to align with fw tuning guide

dagar committed:
Update submodule GPSDrivers to latest Thu Jun 16 12:38:52 UTC 2022

    - GPSDrivers in PX4/Firmware (1069570a2a90fdc7f0e081b6c2c4b418446d65d7): https://github.com/PX4/PX4-GPSDrivers/commit/016c37cd1f18c716427e2465d8daa6aa1054b0f1
    - GPSDrivers current upstream: https://github.com/PX4/PX4-GPSDrivers/commit/8c09c5426d23ea4db4e462c1f4e3a1de33d253cc
    - Changes: https://github.com/PX4/PX4-GPSDrivers/compare/016c37cd1f18c716427e2465d8daa6aa1054b0f1...8c09c5426d23ea4db4e462c1f4e3a1de33d253cc

    8c09c54 2022-06-15 Daniel Agar - sbf trivial whitespace fix

Co-authored-by: PX4 BuildBot <bot@px4.io>

bkueng committed:
mavlink_shell: set target system and component id

These got added in https://github.com/mavlink/mavlink/pull/1725 and need
to be set for correct routing.

AlexKlimaj committed:
uavcannode: Fix dronecan baro units

dagar committed:
sensors: add kconfig mechanism to optionally exclude sensor types

dagar committed:
Tools/kconfig: delete old migration helpers

 - people are updating this without really understanding the purpose,
and it's generally no longer useful

dagar committed:
enable gyro_calibration for CAN nodes

 - allow saving initial gyro cal if very close to 0

dagar committed:
sensors: add CONFIG_SENSORS_VEHICLE_AIRSPEED for airspeed/differential pressure

 - disable CONFIG_SENSORS_VEHICLE_AIRSPEED on boards only used for multicopter

dagar committed:
optical flow sensor pipeline overhaul

 - all sources of optical flow publish sensor_optical_flow
 - sensor_optical_flow is aggregated by the sensors module, aligned with integrated gyro, and published as vehicle_optical_flow

Co-authored-by: alexklimaj <alex@arkelectron.com>

dagar committed:
drivers/optical_flow/paw3902: cleanup/overhaul

 - remove internal accumulation and publish every valid raw sample synchronized with sensor
 - store timestamp_sample from motion interrupt
 - improve timing requirements from datasheet (minimum delays after register read/write)

dagar committed:
drivers/optical_flow/paa3905: cleanup/overhaul

 - remove internal accumulation and publish every valid raw sample synchronized with sensor
 - store timestamp_sample from motion interrupt
 - improve timing requirements from datasheet (minimum delays after register read/write)

AlexKlimaj committed:
Add mag bias estimator to CAN GPS units with IMUs

 - run `sensors` hub to process sensor_mag and publish vehicle_magnetometer
 - update uavcannode to use vehicle_magnetometer

dagar committed:
boards: px4_fmu-v5_protected disable gyro_fft module to save flash

dagar committed:
boards: px4_fmu-v5_test disable fake_gps module to save flash

bmeagher committed:
posix: add mavlink shell for posix targets (#19800)

dagar committed:
boards: px4_fmu-v5_test disable gyro_fft module to save flash

dagar committed:
drivers/differential_pressure/Kconfig: remove ets and ms4515 from common sensors to save flash

dagar committed:
mc_pos_control: always respect position estimate vxy_max if set

Co-authored-by: Matthias Grob <maetugr@gmail.com>

dagar committed:
ekf2: optical flow control limits constrain speed using HAGL max

cmic0 committed:
esc_battery: account for online ESCs when averaging voltage

Signed-off-by: Claudio Micheli <claudio@auterion.com>

faberc committed:
drivers: Sagetech MXS transponder support

Co-authored-by: Megan McCormick <megan.mccormick@sagetech.com>
Co-authored-by: Chuck Faber <chuck.faber@sagetech.com>

dagar committed:
boards: px4_fmu-v5_uavcanv0periph disable modules to save flash

dagar committed:
paw3902: update scaling from datasheet

dagar committed:
paa3905: update scaling from datasheet

Commit:
cuav-nora: changed brick to compatible with cuav noraplus

ChrisRuw committed:
removed unused code - _constrainOneSide and _constrainAbs

MaEtUgR committed:
examples: remove empty fake_gyro

MaEtUgR committed:
uuv_pos_control: remove practically unused manual control subscription

MaEtUgR committed:
examples/fixedwing_control: use initializers instead of memset

MaEtUgR committed:
fw/uuv control: remove duplicated comments, restyle initializers
